10_rest/runnerdave/pkg/puppy/rest_test.go

```diff
@@ -0,0 +1,224 @@
+package puppy_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	puppy "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy"
+	store "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy/store"
+)
+
+var (
+	rs puppy.RestServer
+)
+
+func TestGetPuppy(t *testing.T) {
+	req, err := http.NewRequest("GET", "/api/puppy/1", nil)
+	if err != nil {
```

Comment:
Since you have imported `stretchr/testify/require`, I suggest you use that to verify test setup:
```
require.NoError(err)
```
That will end the test there if there is an error, and it takes 1 line instead of three.